cpop_bot/music_grabber.py

```diff
@@ -17,102 +17,101 @@ class WrongFileSizeError(ValueError):
-        audio_file = YoutubeDL(ydl_opts).prepare_filename(info)
-        thumbnail_file = audio_file.rsplit(".", 1)[-2] + "." + \
-            self.__get_file_extension_from_url(thumbnail_url)
-        info['downloads'] = {
-            'audio': audio_file,
-            'thumbnail': thumbnail_file
-        }
-        return info
-
-    def __files_successfully_downloaded(self, info_downloads):
-        return all(os.path.isfile(f) for f in info_downloads.values())
-
-    def __delete_downloaded_files(self, info_downloads):
-        for f in info_downloads.values():
-            try:
-                os.remove(f)
-            except OSError:
-                pass
-
-    def __get_file_extension_from_url(self, url):
-        url_path = urlparse(url).path
-        basename = os.path.basename(url_path)
-        return basename.split(".")[-1]
-
-
-class SquarethumbMaker:
-
-    # https://stackoverflow.com/a/52177551
-    def make_squarethumb(self, thumbnail, output):
-        original_thumb = Image.open(thumbnail)
-        squarethumb = self.__crop_to_square(original_thumb)
-        squarethumb.thumbnail((320, 320), Image.ANTIALIAS)
-        squarethumb.save(output)
-
-    def __crop_to_square(self, img):
-        width, height = img.size
-        length = min(width, height)
-        left = (width - length) / 2
-        top = (height - length) / 2
-        right = (width + length) / 2
-        bottom = (height + length) / 2
-        return img.crop((left, top, right, bottom))
+def download_audio(input_url, filesize_limit):
+    input_url = input_url.split('&list', 1)[0]
+    ydl_opts = {
+        'format': 'bestaudio[protocol^=http]',
+        'logger': logger,
+        'outtmpl': DOWNLOAD_DIR
+        + '%(title)s - %(extractor)s-%(id)s.%(ext)s',
+        'writethumbnail': True
+    }
+    ydl = YoutubeDL(ydl_opts)
+    info = ydl.extract_info(input_url, download=False)
+    webpage_info = info['extractor'] + " " + info['id']
+    info_format = next((item for item in info['formats']
+                        if item['format_id'] == info['format_id']), None)
+
+    if youtube_video_not_music(info):
+        raise WrongCategoryError(webpage_info + ": not under Music category")
+
+    audio_filesize = get_filesize(info_format)
+    if int(audio_filesize) > filesize_limit:
+        raise WrongFileSizeError(webpage_info + ": audio file size ("
+                                 + str(audio_filesize) + ") is greater"
+                                 + "than the limit: (" + str(filesize_limit)
+                                 + ")")
+
+    logger.info("%s: downloading...", webpage_info)
+    ydl.process_info(info)
+    info = add_downloads_to_info_dict(info, ydl_opts)
+    if not files_successfully_downloaded(info['downloads']):
+        delete_downloaded_files(info['downloads'])
+        raise ValueError(webpage_info
+                         + ": failed to download audio or thumbnail")
+    return info
+
+
+def youtube_video_not_music(info):
```

Comment:
huh so there's no private functions inside the file in python, interesting

Reply:
I think private methods `__method_name` are also not actually private in python, those are just name mangling. but I'm not sure about this.

> “Private” instance variables that cannot be accessed except from inside an object don’t exist in Python. However, there is a convention that is followed by most Python code: a name prefixed with an underscore (e.g. _spam) should be treated as a non-public part of the API (whether it is a function, a method or a data member). It should be considered an implementation detail and subject to change without notice.

https://docs.python.org/3/tutorial/classes.html#private-variables

I think we should use `_function_name` here. I'm not sure if I misunderstand anything.

Reply:
Pushed new commit which does this.